spec/models/voucher_spec.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Voucher do
+  describe 'associations' do
+    it { is_expected.to belong_to(:enterprise) }
+  end
+
+  describe 'validations' do
+    subject { Voucher.new(code: 'new_code', enterprise: enterprise) }
+
+    let(:enterprise) { build(:enterprise) }
+
+    it { is_expected.to validate_presence_of(:code) }
+    it { is_expected.to validate_uniqueness_of(:code).scoped_to(:enterprise_id) }
```

Comment:
I was surprised that you defined the subject because most shoulda matchers don't need that. It's only the uniqueness which needs that. So strictly speaking, the presence test doesn't need the subject and we don't need to build that enterprise. But since you are only building it in memory and these specs execute in less than a second it doesn't matter. Good solution.

Reply:
How would you do it then ? Would you just build the subject for the uniqueness `it` block ?

Reply:
Yes, I would have defined it in there:

```rb
it "validates code uniqueness per enterprise" do
  subject = Voucher.new(code: 'new_code', enterprise: enterprise)
  expect(subject).to validate_uniqueness_of(:code).scoped_to(:enterprise_id)
end
```

But I don't think it matters here. Your solution is more elegant in general and looks better. Maybe it's good to always supply valid values to the matchers. It provides an example and doesn't rely on Shoulda to generate random valid values. :shrug: 

I think that my example above would only be better if it improved the execution time significantly.